BendingSpoons/fastlane
# [FEAT] Metadata improvements

### Checklist

### Motivation and Context


Addresses robustness issues with the upload of visual metadata to App Store Connect and adds experimental deliver support for uploading app previews.

### Description


The `upload_screenshots.rb` file has been revamped to support better parallelisation and API retry methods with respect to the upstream fastlane implementation. In addition, a new `upload_app_previews.rb` file has been added to offer a high level interface for app previews uploads. 

At the moment, `upload_app_previews.rb` is not a full mirror of `upload_screenshots.rb`, as a few features are missing:

- There's no automatic detection and validation of app previews sizes
- The upload method can't be invoked with _deliver_, but only with a ruby api call
- There's no automatic collection of app previews

### Testing Steps


The update has been tested live with a bunker API call. Existing metadata-related tests still pass, but ideally new rspec tests should be written.
